### ISSUE TITLE
async-generator - why required in Python 3.9?

### ISSUE THREAD

**Issue by edrozenberg** [jupyter/nbclient#133]
`Python 3.9.1`

Running `pipdeptree` I see:
```
* nbclient==0.5.2
 - async-generator [required: Any, installed: ?]
 ```

But according to https://github.com/python-trio/async_generator "This library gives you all that back to Python 3.5." which implies python 3.6+ already have async generator functionality.

Is it because `nbclient` is using `async-generator` on all versions of python3 for consistency?

**Comment by davidbrochart** [comment]
Good point, I think we could get rid of it, since we don't support Python 3.5 anymore.

**Comment by davidbrochart** [comment]
Actually we need `async-generator` for `asynccontextmanager`, which is only available from Python 3.7.

**Comment by edrozenberg** [comment]
OK I understand, thank you.
